modules/apps/29-fee/types/keys_test.go

```diff
@@ -139,3 +139,40 @@ func TestParseKeyForwardRelayerAddress(t *testing.T) {
 		}
 	}
 }
+
+func TestParseKeyCounterpartyRelayer(t *testing.T) {
+	var (
+		relayerAddress             = "relayer_address"
+		counterpartyRelayerAddress = "counterparty_address"
+	)
+
+	testCases := []struct {
+		name    string
+		key     string
+		expPass bool
+	}{
+		{
+			"success",
+			string(types.KeyCounterpartyRelayer(relayerAddress, ibctesting.FirstChannelID)),
+			true,
+		},
+		{
+			"incorrect key - key split has incorrect length",
+			"relayerAddress/relayer_address/transfer/channel-0",
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		address, channelID, err := types.ParseKeyCounterpartyRelayer(tc.key)
+		validRegisteredAddr := types.RegisteredRelayerAddress{relayerAddress, counterpartyRelayerAddress, ibctesting.FirstChannelID}
```

Comment:
This could probably be removed now and you could do the require checks using the channelID and relayerAddress directly